test/e2e/run-all.js

```diff
@@ -212,12 +213,26 @@ async function main() {
 
   console.log('My test list:', myTestList);
 
+  const changedOrNewTests = await fetchChangedE2eFiles();
+  console.log('Spec files that will be re-run:', changedOrNewTests);
+
   // spawn `run-e2e-test.js` for each test in myTestList
   for (let testPath of myTestList) {
     if (testPath !== '') {
       testPath = testPath.replace('\n', ''); // sometimes there's a newline at the end of the testPath
       console.log(`\nExecuting testPath: ${testPath}\n`);
-      await runInShell('node', [...args, testPath]);
+
+      const testFileName = testPath.split('/').pop();
+      const isTestChangedOrNew = changedOrNewTests.includes(testFileName);
+      const retryIndex = args.indexOf('--retries');
+      if (retryIndex !== -1) {
+        args.splice(retryIndex, 2);
+      }
+
+      const extraArgs = isTestChangedOrNew
+        ? ['--retry-until-failure', '--retries=5']
```

Comment:
the number 5 is arbitrary, we could change it if another number is better suited

Reply:
Shall we extract it to a const and then it is easier to make modifications in the future

Reply:
thank you for your suggestion @DDDDDanica :bowing_woman: I've moved this into a const :+1: